std/experimental/allocator/package.d

```diff
@@ -1731,17 +1731,18 @@ reference, or an entire array. It is assumed the respective entities had been
 allocated with the same allocator.
 
 */
-void dispose(A, T)(auto ref A alloc, T* p)
+void dispose(A, T)(auto ref A alloc, auto ref T* p)
 {
     static if (hasElaborateDestructor!T)
     {
         destroy(*p);
     }
     alloc.deallocate((cast(void*) p)[0 .. T.sizeof]);
+    p = null;
```

Comment:
`static if (is(p == ref)) p = null;`

Reply:
I guess you mean https://dlang.org/spec/traits.html#isRef, since the `is` expression works only with types and `ref` is not a type qualifier.

Also `int x; return is(x == int)` returns false which is a bit surprising. Probably worth filing a Bugzilla issue?

Reply:
```d
static if ((ParameterStorageClassTuple!dispose)[1] ==
            ParameterStorageClass.ref_) p = null;
```

That would be that which is much for just a simple assignment.

Reply:
@ZombineDev  Yes, `__traits(isRef, p)` is needed, sorry.

You need `is(typeof(x) == int)` there. 

@BBasile please try `__traits(isRef, p)`

Reply:
> You need is(typeof(x) == int) there.

@andralex yes, I'm well aware, I'm just pointing out that the result from my example would look strange to the uninitiated, so I was wondering if the compiler should emit a warning or error, in order to prevent accidental misuse.

Reply:
@ZombineDev I see. Problem is, sometimes you _do_ want to check whether an alias is a type...